pkg/controllers/deprovisioning/helpers.go

```diff
@@ -147,6 +149,128 @@ func disruptionCost(ctx context.Context, pods []*v1.Pod) float64 {
 	return cost
 }
 
+// candidateNodes returns nodes that appear to be currently deprovisionable based off of their provisioner
+// nolint:gocyclo
+func candidateNodes(ctx context.Context, cluster *state.Cluster, kubeClient client.Client, clk clock.Clock, cloudProvider cloudprovider.CloudProvider,
```

Comment:
nit: It feels like candidateNodes is a core part of the controller so it feels weird that this is in the helpers file

Reply:
While this in the main deprovisioning loop, it's also used in each of the deprovisioners that implement replacement. Likewise to the simulateScheduling, I think it fits best here. WDYT?